packages/boot/test/upgrading/vat-vow.js

```diff
@@ -0,0 +1,65 @@
+import { prepareVowTools } from '@agoric/vat-data/vow.js';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { makePublishKit, subscribeLatest } from '@agoric/notifier';
+import { Far } from '@endo/far';
+
+export const buildRootObject = (_vatPowers, _args, baggage) => {
+  const zone = makeDurableZone(baggage);
+  const { watch } = prepareVowTools(zone.subZone('VowTools'));
+
+  /** @type {PublishKit<null>} */
+  const { publisher, subscriber } = makePublishKit();
+  publisher.publish(null);
+
+  let numRejected = 0n;
+  let numFulfilled = 0n;
+  const makeWatcher = zone.exoClass(
+    'PromiseWatcher',
+    undefined,
+    (expectFulfilled, value) => ({ expectFulfilled, value }),
+    {
+      async onFulfilled(_value) {
+        // console.log('fulfilled:', value, { ...this.state });
+        numFulfilled += 1n;
+        publisher.publish(null);
+      },
+      async onRejected(_reason) {
+        // console.log('rejected:', reason, { ...this.state });
+        numRejected += 1n;
+        publisher.publish(null);
+      },
+    },
+  );
+
+  return Far('VowRoot', {
+    async testV0() {
+      // Watch a promise that isn't resolved in this incarnation.
+      const pForever = new Promise(() => {});
+      watch(pForever, makeWatcher());
+
+      // Watch one that is resolved.
+      watch(Promise.resolve('hello'), makeWatcher(true, 'hello'));
+
+      for await (const _ of subscribeLatest(subscriber)) {
+        if (numRejected > 0n) {
+          throw new Error('unexpected rejections');
+        }
+        if (numFulfilled > 0n) {
+          break;
+        }
+      }
```

Comment:
I find this style to be very opaque; can we instead put the test logic in packages/boot/test/upgrading/test-upgrade-vats.js?
```js
  const vowRoot = await EV.vat('bootstrap').createVat(vowVatConfig);

  t.log('test incarnation 0');
  /** @type {Record<string, [settlementValue?: unknown, isRejection?: boolean]>} */
  const localPromises = { forever: [], fulfilled: ['hello'] };
  await EV(vowRoot).makeLocalPromiseWatchers(localPromises);
  t.deepEqual(await EV(vowRoot).getWatcherResults(), {
    fulfilled: { status: 'fulfilled', value: 'hello' },
    forever: { status: 'unsettled' },
  });

  t.log('restart');
  const { incarnationNumber } =
    await EV.vat('bootstrap').upgradeVat(vowVatConfig);
  t.is(incarnationNumber, 1, 'vat must be reincarnated');

  t.log('test incarnation 1');
  t.deepEqual(await EV(vowRoot).getWatcherResults(), {
    fulfilled: { status: 'fulfilled', value: 'hello' },
    forever: {
      status: 'rejected',
      reason: {
        name: 'vatUpgraded',
        upgradeMessage: 'vat upgraded',
        incarnationNumber: 0,
      },
    },
  });
```